framework/src/base/NonlinearEigenSystem.C

```diff
@@ -30,10 +30,20 @@ namespace Moose {
 #if LIBMESH_HAVE_SLEPC
 void assemble_matrix(EquationSystems & es, const std::string & system_name)
 {
-  FEProblemBase * p = es.parameters.get<FEProblemBase *>("_fe_problem_base");
+  EigenProblem * p = es.parameters.get<EigenProblem *>("_eigen_problem");
   EigenSystem & eigen_system = es.get_system<EigenSystem>(system_name);
 
   p->computeJacobian(*eigen_system.current_local_solution.get(), *eigen_system.matrix_A);
+
+  // if it is a generalized eigenvalue problem
+  bool generalized_eigenvalue_problem = p->isGeneralizedEigenvalueProblem();
+  if (generalized_eigenvalue_problem)
+  {
+    if (eigen_system.matrix_B)
+      p->computeJacobian(*eigen_system.current_local_solution.get(), *eigen_system.matrix_B);
+    else
+      mooseError("It is a generalized eigenvalue problem but matrix B is empty \n");
```

Comment:
So at this point we have to explicitly assemble B and no B-matrix-free is supported?